src/providers/ldap/ldap_id_enum.c

```diff
@@ -114,14 +115,22 @@ errno_t ldap_id_setup_enumeration(struct be_ctx *be_ctx,
     if (ret != EOK) {
         DEBUG(SSSDBG_FATAL_FAILURE,
               "Unable to initialize enumeration periodic task\n");
-        talloc_free(ectx);
-        return ret;
+        goto fail;
     }
 
     talloc_steal(sdom->enum_task, ectx);
     talloc_free(name);
 
     return EOK;
+
+fail:
+    if (name != NULL) {
+        talloc_free(name);
+    }
+    if (ectx != NULL) {
+        talloc_free(ectx);
+    }
+    return ret;
```

Comment:
`talloc_free()` can handle `NULL` so the NULL-check is not needed. Also it would be better to use `goto done` patter here instead of `goto fail`, i.e.:

```c
     if (!success) {
         ret = EFAULT;
         goto done;
     }

     ret = EOK;

done:
    if (ret != EOK) {
          talloc_free(name);
          talloc_free(ectx);
    }

    return ret;
```